SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialLatencyTracer.h"
+
+#include "Async/Async.h"
+#include "Engine/World.h"
+#include "EngineClasses/SpatialGameInstance.h"
+#include "Interop/Connection/OutgoingMessages.h"
+#include "Utils/SchemaUtils.h"
+
+#include <sstream>
+
+DEFINE_LOG_CATEGORY(LogSpatialLatencyTracing);
+
+// Stream for piping trace lib output to UE output
+class UEStream : public std::stringbuf
+{
+	int sync()
+	{
+		UE_LOG(LogSpatialLatencyTracing, Verbose, TEXT("%s"), *FString(str().c_str()));
+		str("");
+		return std::stringbuf::sync();
+	}
+};
+
+namespace
+{
+	UEStream Stream;
+}
+
+const TraceKey USpatialLatencyTracer::ActiveTraceKey = 0;
```

Comment:
I've found these confusing to read through. A comment saying what I'm expecting with these would be useful. Might be something to discuss offline.

Reply:
I'll add a comment